search.php

```diff
@@ -9,10 +9,92 @@
  * @since   Timber 0.1
  */
 
+/**
+ * Planet4 - Search functionality.
+ */
+
+global $wp_query;
+
 $templates = array( 'search.twig', 'archive.twig', 'index.twig' );
-$context = Timber::get_context();
+$context   = Timber::get_context();
+$context['sort_options'] = [
+	'relevant' => __( 'Most relevant', 'planet4-master-theme' ),
+	'recent'   => __( 'Most recent', 'planet4-master-theme' ),
+];
+
+$default_sort  = 'relevant';
+$selected_sort = filter_input( INPUT_GET, 'order', FILTER_SANITIZE_STRING );
+$search        = get_search_query();
+
+if ( ! in_array( $selected_sort, array_keys( $context['sort_options'] ), true ) ) {
+	$context['selected_sort'] = $default_sort;
+} else {
+	$context['selected_sort'] = $selected_sort;
+}
+
+switch ( $context['selected_sort'] ) {
+	case 'recent':
+		$context['posts'] = Timber::get_posts( [
+			// TODO - Find solution for Timber bug (see https://github.com/timber/timber/issues/935).
+			// which does not include attachments to get_posts() results if we supply an array as an argument.
+			's'       => $search,
```

Comment:
Looks like this is unrelated to Timber assuming you're trying to get attachments returning as top level items in the result set. 
https://github.com/WordPress/WordPress/blob/master/wp-includes/class-wp-query.php#L2323

Attachments have a special 'inherit' post status which means they take the status from their parent post. I hadn't realised this before but this looks like this status only gets handled transparently by core if the is_tax is set to true on the query object. Unless there needs to be a concept of unpublished attachments on site, we can just add the 'inherit' status to our post_status arg 

```
'post_status' => ['publish', 'inherit'],
'post_type' => ['post', 'attachment']
```

Otherwise need to set is_tax to true (needs testing) or filtering `posts_where` and `posts_join` might be the best option.
https://github.com/WordPress/WordPress/blob/master/wp-includes/class-wp-query.php#L2376

Reply:
@tcrsavage 
Hi Theo, I had already tried the 'inherit' post_status option, does not work. Also, tried with 'any'.

Can you explain more what you mean with the is_tax ? Will try to use the posts_where, posts_join filters, see what I can get.

Reply:
@tcrsavage 
Looks like this might have to do with Timber and SearchWP not collaborating very well.
If I don't pass a parameter to Timber::get_posts it works as expected with SearchWP...
but if I do pass a parameter then it looks like it skips SearchWP somehow.

Reply:
@tcrsavage @koyan 
Looks like this might have to do with Timber and SearchWP not collaborating very well.
If I don't pass a parameter to Timber::get_posts it works as expected with SearchWP...
but if I do pass a parameter then it looks like it skips SearchWP somehow.

Reply:
@angtheod will ping you to help with some debugging on this. I'm able to get attachments pulling in on my local with 'inherit' status but haven't dug into the SearchWP side of things yet.